src/main/java/io/netty/incubator/codec/http3/QpackEncoderDynamicTable.java

```diff
@@ -0,0 +1,448 @@
+        }
+        if (e == toRemove) {
+            if (prev == null) {
+                fields[i] = e.nextSibling;
+            } else {
+                prev.nextSibling = e.nextSibling;
+            }
+        }
+
+        // Remove from the linked list
+        toRemove.remove(head);
+        if (toRemove == tail) {
+            resetIndicesToHead();
+        }
+        if (toRemove == drain) {
+            drain = head;
+        }
+        if (toRemove == knownReceived) {
+            knownReceived = head;
+        }
+        return true;
+    }
+
+    private void resetIndicesToHead() {
+        tail = head;
+        drain = head;
+        knownReceived = head;
+    }
+
+    private void ensureFreeCapacity() {
+        long maxDesiredSize = max(ENTRY_OVERHEAD, ((100 - expectedFreeCapacityPercentage) * capacity) / 100);
+        long cSize = size;
+        HeaderEntry nDrain;
+        for (nDrain = head; nDrain.next != null && cSize > maxDesiredSize; nDrain = nDrain.next) {
+            cSize -= nDrain.next.size();
+        }
+        if (cSize != size) {
+            drain = nDrain;
+            evictUnreferencedEntries();
+        }
+    }
+
+    private int index(int h) {
+        return h & hashMask;
+    }
+
+    private static void validateCapacity(long capacity) throws QpackException {
+        if (capacity < MIN_HEADER_TABLE_SIZE || capacity > MAX_HEADER_TABLE_SIZE) {
+            throw INVALID_TABLE_CAPACITY;
+        }
+    }
+
+    /**
+     * An entry for the {@link #fields} HashMap. This entry provides insertion order iteration using {@link #next}.
+     */
+    private static final class HeaderEntry extends QpackHeaderField {
+        /**
+         * Pointer to the next entry in insertion order with a different {@link #hash} than this entry.
+         */
+        HeaderEntry next;
+
+        /**
+         * Pointer to the next entry in insertion order with the same {@link #hash} as this entry, a.k.a hash collisions
+         */
+        HeaderEntry nextSibling;
+
+        /**
+         * Number of header blocks that refer to this entry as the value for its <a
+         * href="https://quicwg.org/base-drafts/draft-ietf-quic-qpack.html#name-required-insert-count">
+         * required insert count</a>
+         */
+        int refCount;
+
+        /**
+         * Hashcode for this entry.
+         */
+        final int hash;
+
+        /**
+         * Insertion index for this entry.
+         */
+        final int index;
+
+        HeaderEntry(int hash, CharSequence name, CharSequence value, int index, HeaderEntry nextSibling) {
+            super(name, value);
+            this.index = index;
+            this.hash = hash;
+            this.nextSibling = nextSibling;
+        }
+
+        void remove(HeaderEntry before) {
+            before.next = next;
+            next = null; // null references to prevent nepotism in generational GC.
+            nextSibling = null;
+        }
+
+        void addNextTo(HeaderEntry prev) {
```

Comment:
should we make the naming consistent and either use `prev` in the above method as well or `before` here ?